core/src/main/java/dk/alexandra/fresco/framework/util/StrictBitVector.java

```diff
@@ -0,0 +1,199 @@
+      throw new IllegalArgumentException("Size must be multiple of 8");
+    }
+    this.size = size;
+    this.bits = new byte[size / 8];
+  }
+
+  /**
+   * Constructs new strict bit vector from a secure source of randomness.
+   * 
+   * @param bits raw bytes
+   * @param rand secure source of randomness
+   */
+  public StrictBitVector(int size, Drbg rand) {
+    this(size);
+    rand.nextBytes(bits);
+  }
+
+  /**
+   * Returns the "bit" number bit, reading from left-to-right, from a byte array.
+   * 
+   * @param input The arrays of which to retrieve a bit
+   * @param bit The index of the bit, counting from 0
+   * @param isBigEndian Indicates whether the underlying byte array should be interpreted as
+   *        big-endian or little-endian
+   * @return Returns the "bit" number bit, reading from left-to-right, from "input"
+   */
+  public boolean getBit(int bit, boolean isBigEndian) {
+    rangeCheck(bit);
+    int actualIndex = isBigEndian ? size - 1 - bit : bit;
+    return ByteArrayHelper.getBit(bits, actualIndex);
+  }
+
+  /**
+   * Sets the bit at index to value, reading from left-to-right.
+   * 
+   * @param index index of the bit to be set
+   * @param value value to set bit to
+   * @param isBigEndian Indicates whether the underlying byte array should be interpreted as
+   *        big-endian or little-endian
+   */
+  public void setBit(int index, boolean value, boolean isBigEndian) {
+    rangeCheck(index);
+    int actualIndex = isBigEndian ? size - 1 - index : index;
+    ByteArrayHelper.setBit(bits, actualIndex, value);
+  }
+
+  /**
+   * Returns the "bit" number bit, reading from left-to-right, from a byte array. <br>
+   * Assumes underlying byte array is big-endian.
+   * 
+   * @param input The arrays of which to retrieve a bit
+   * @param bit The index of the bit, counting from 0
+   * @return Returns the "bit" number bit, reading from left-to-right, from "input"
+   */
+  public boolean getBit(int bit) {
+    return getBit(bit, true);
+  }
+
+  /**
+   * Sets the bit at index to value, reading from left-to-right. <br>
+   * Assumes underlying byte array is big-endian.
+   * 
+   * @param index index of the bit to be set
+   * @param value value to set bit to
+   */
+  public void setBit(int index, boolean value) {
+    setBit(index, value, true);
+  }
+
+  public int getSize() {
+    return size;
+  }
+
+  public byte[] toByteArray() {
+    return bits.clone();
+  }
+
+  /**
+   * Concatenates bit vectors into one bit vector.
+   * 
+   * @param reverse
+   * @param bitVectors
+   * @return
+   */
+  public static StrictBitVector concat(boolean reverse, StrictBitVector... bitVectors) {
+    if (reverse) {
+      Collections.reverse(Arrays.asList(bitVectors));
+    }
+    // compute length of result byte array and number of bits
+    int combinedByteLength = 0;
+    int combinedBitLength = 0;
+    for (StrictBitVector bitVector : bitVectors) {
+      combinedByteLength += bitVector.getSize() / 8;
```

Comment:
Will this always work? Are we guaranteed that the array holds a multiple of 8 bits? If there is only e.g. 7, then the byte size will be 0 according to this. Intentional? Just wanted to be sure :)

Reply:
Yep, `StrictBitVector` instances are guaranteed to have a bit length that is a multiple of 8.